pkgs/swiftgen/swift2objc/lib/src/parser/parsers/parse_declarations_map.dart

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:developer';
+
+import 'package:swift2objc/src/ast/_core/interfaces/declaration.dart';
+import 'package:swift2objc/src/parser/_core/parsed_symbolgraph.dart';
+import 'package:swift2objc/src/parser/parsers/declaration_parsers/parse_class_decalartion.dart';
+import 'package:swift2objc/src/parser/parsers/declaration_parsers/parse_method_declaration.dart';
+
+import '../_core/utils.dart';
+
+DeclarationsMap parseDeclarationsMap(ParsedSymbolgraph symbolgraph) {
```

Comment:
It looks like you're not really using the declarations map as a map. You build the map here and then just turn it into a list in parseAst. You could just return a list here.

And if you do that, you can iterate through `symbolgraph.symbols.values` rather than `keys`, which is a bit cleaner.

Reply:
ur absolutely right. Didn't notice that it's not being used as a map anymore.

Done :)